scripts/inference.py

```diff
@@ -114,6 +128,9 @@
         distr_param = "mp"
     else:
         distr_param = None
+if args.distributed_strategy:
+    # --distributed_strategy option overrides --distributed option
+    distr_param = args.distributed_strategy
```

Comment:
if we're going to pass a distributed_stategy, I think the --distributed arg becomes redundant (equivalent to `args.distributed_strategy is not None`)

Reply:
Is it ok to remove `--distributed`? I intentionally leave it for compatibility while `--distributed` will become redundant.

Reply:
I think it depends on the goal / what we're trying to achieve.

The potential distribution paths are:

* mp -- this needs to be explicitly requested since it's used with worldsize=1 (without init_process_group) and can't be inferred.
* tp -- this is assumed to be used whenever worldsize > 1
* fsdp -- we're currently not doing inference with fsdp b/c it's a slower and it's an atypical usage pattern. if we do want both fsdp and tp, they can't be inferred implicitly.
* None -- no distribution

As of right now, the user only needs to specify when/if they want MP (--distributed=True). If we add distributed_strategy in place of --distributed, it does seem to impose a bit more on the user since they'll need to specify it for tp, when they wouldn't have had to otherwise.

If the goal / use-case is to support fsdp inference in this script, then it makes sense that we would need a more granular distributed_strategy parameter.

There's also a potential argument for consistency, but I'm not sure if that really matters here (if not changing benchmark_inference).

I think it would be confusing to keep both parameters.

Also btw, if we do change the usage, we'll want to change the example usage at the top of the script too.